packages/@tinacms/cli/src/cmds/init/apply.ts

```diff
@@ -0,0 +1,721 @@
+
+  logNextSteps({
+    packageManager: config.packageManager,
+    framework: config.framework,
+  })
+}
+
+const forestryMigrate = async ({
+  pathToForestryConfig,
+  usingTypescript,
+  frontMatterFormat,
+}: {
+  usingTypescript: boolean
+  pathToForestryConfig: string
+  frontMatterFormat: 'yaml' | 'toml' | 'json'
+}) => {
+  const { collections, importStatements, templateCode } =
+    await generateCollections({
+      pathToForestryConfig,
+      usingTypescript,
+      frontMatterFormat,
+    })
+
+  // print errors
+  // This error is handled now so we do not need to print it
+  // ErrorSingleton.getInstance().printCollectionNameErrors()
+  const JSONString = JSON.stringify(collections, null, 2)
+
+  const { code } = addVariablesToCode(JSONString)
+
+  return {
+    collectionString: code,
+    importStatements,
+    templateCodeString: templateCode,
+  }
+}
+
+const reportTelemetry = async ({
+  hasForestryConfig,
+  noTelemetry,
+  usingTypescript,
+}: {
+  usingTypescript: boolean
+  noTelemetry: boolean
+  hasForestryConfig: boolean
+}) => {
+  if (noTelemetry) {
+    logger.info(logText('Telemetry disabled'))
+  }
+  const telemetry = new Telemetry({ disabled: noTelemetry })
+  const schemaFileType = usingTypescript ? 'ts' : 'js'
+  await telemetry.submitRecord({
+    event: {
+      name: 'tinacms:cli:init:invoke',
+      schemaFileType,
+      hasForestryConfig,
+    },
+  })
+}
+
+const createPackageJSON = async () => {
+  logger.info(logText('No package.json found, creating one'))
+  await execShellCommand(`npm init --yes`)
+}
+const createGitignore = async ({ baseDir }: { baseDir: string }) => {
+  logger.info(logText('No .gitignore found, creating one'))
+  await fs.outputFileSync(
+    path.join(baseDir, '.gitignore'),
+    'node_modules\n.env.tina'
+  )
+}
+
+const updateGitIgnore = async ({
+  baseDir,
+  items,
+}: {
+  baseDir: string
+  items: string[]
+}) => {
+  logger.info(logText(`Adding ${items.join(',')} to .gitignore`))
+  const gitignoreContent = fs
+    .readFileSync(path.join(baseDir, '.gitignore'))
+    .toString()
+  const newGitignoreContent = [...gitignoreContent.split('\n'), ...items].join(
+    '\n'
+  )
+  await fs.writeFile(path.join(baseDir, '.gitignore'), newGitignoreContent)
+}
+const addDependencies = async (
+  config: Record<any, any>,
+  env: InitEnvironment
+) => {
+  const { dataLayer, dataLayerAdapter, nextAuth, packageManager } = config
+  logger.info(logText('Adding dependencies, this might take a moment...'))
+  const tailwindDeps = [
+    '@tailwindcss/typography',
+    'tailwindcss',
+    'postcss',
+    'autoprefixer',
+  ]
+  const deps = ['tinacms', '@tinacms/cli']
+  const devDeps = []
+  if (nextAuth) {
+    deps.push('tinacms-next-auth', 'next-auth')
+    if (config.installTailwindCSS && !env.tailwindConfigExists) {
+      devDeps.push(...tailwindDeps)
+    }
+  }
+  if (dataLayer) {
+    deps.push('@tinacms/datalayer')
+    deps.push('@octokit/rest')
```

Comment:
@octokit/rest and js-base64 are deps of `tinacms-gitprovider-github` so they are not needed to install.